zerver/actions/realm_settings.py

```diff
@@ -552,6 +561,13 @@ def do_deactivate_realm(
         event = dict(type="realm", op="deactivated", realm_id=realm.id)
         send_event_on_commit(realm, event, active_user_ids(realm.id))
 
+        if deletion_delay_days == 0:
+            deferred_work_event = {
+                "type": "scrub_deactivated_realm",
+                "realm_id": realm.id,
+            }
+            queue_json_publish("deferred_work", deferred_work_event)
+
```

Comment:
This was suggested [over here](https://github.com/zulip/zulip/pull/26109#discussion_r1242558033) by @alexmv.